tripy/tripy/frontend/trace/ops/binary_elementwise.py

```diff
@@ -70,9 +70,7 @@ def infer_shape_output_idxs(self, inputs):
                     ]
                 )
             return Result.ok({"shape": [0]})
-        elif any(
-            map(lambda t: isinstance(t, ShapeScalar) or (isinstance(t, int) and not isinstance(t, Tensor)), inputs)
-        ):
+        elif all(map(lambda t: isinstance(t, ShapeScalar), inputs)):
```

Comment:
Does not need to be done in this PR, but we might want to rename `infer_shape_output_idxs` since it's now being used to make judgments about `ShapeScalar`s too. We could fold that into discussion of #233

Reply:
Mentioned in the issue.